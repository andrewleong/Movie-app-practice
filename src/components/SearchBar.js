import React from 'react';

const SearchBarComponent = ({ onQueryChange, onSearch, disabled }) => {
    return (
        <div className="search-bar">
            <input id="search-bar" onChange={onQueryChange} type="text" placeholder="Search Movies" />
            <button type="button" disabled={disabled} onClick={onSearch} className="btn btn-primary">Search</button>
        </div>
    );
};

export default SearchBarComponent;
