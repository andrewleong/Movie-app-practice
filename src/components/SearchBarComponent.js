import React from 'react';

const SearchBarComponent = ({ onQueryChange }) => {
    return (
        <div>
            <input id="search-bar" onChange={onQueryChange} type="text" placeholder="Search Movies" />
        </div>
    );
};

export default SearchBarComponent;
