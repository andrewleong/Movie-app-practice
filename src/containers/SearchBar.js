import React, { Component } from 'react';

class SearchBar extends Component {
    constructor(props, context) {
        super(props, context);
        this.state = {

        }
    }

    render() {
        return (
            <h1>Search bar test</h1>
        );
    }
}

export default SearchBar;
