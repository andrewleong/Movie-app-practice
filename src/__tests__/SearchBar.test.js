import React from 'react';
import { shallow, mount } from 'enzyme';
import SearchBarComponent from '../components/SearchBar';
import SearchBar from '../containers/SearchBar';



describe('(Component) Search Bar', () => {
    const wrapper = shallow(<SearchBarComponent />);

    it('should render an <input /> to type movie names', () => {
        expect(wrapper.find('input')).toHaveLength(1);
    });

    it('should render a <button /> to search for movies', () => {
        expect(wrapper.find('button')).toHaveLength(1);
    });

    it('should call onQueryChange while typing', () => {
        const mockFn = jest.fn();
        const value = "inception";
        const wrapper = shallow(
            <SearchBarComponent
                onQueryChange={mockFn}
            />
        )

        wrapper.find('#search-bar').simulate('change', null, value)
        expect(mockFn).toBeCalledWith(null, value);
        expect(wrapper.props()).toMatchSnapshot();
    })

    it('should be able to click search button', () => {
        const mockFn = jest.fn();
        const wrapper = shallow(
            <SearchBarComponent
                onSearch={mockFn}
            />
        )

        wrapper.find('button').simulate('click')
        expect(mockFn).toHaveBeenCalled();
        expect(wrapper.props()).toMatchSnapshot();
    })

});


describe('(Container) Search Bar', () => {
    it('should be able to click search when button is disabled', () => {
        const mockFn = jest.fn();
        const wrapper = mount(
            <SearchBar
                onSearch={mockFn}
            />
        )

        wrapper.setState({ disabled: true });

        wrapper.find('button').simulate('click')
        expect(mockFn).not.toHaveBeenCalled();
    })
})


