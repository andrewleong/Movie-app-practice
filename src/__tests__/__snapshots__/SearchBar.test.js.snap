// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Search Bar should be able to click search button 1`] = `
Object {
  "children": Array [
    <input
      id="search-bar"
      placeholder="Search Movies"
      type="text"
    />,
    <button
      className="btn btn-primary"
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      type="button"
    >
      Search
    </button>,
  ],
  "className": "search-bar",
}
`;

exports[`(Component) Search Bar should call onQueryChange while typing 1`] = `
Object {
  "children": Array [
    <input
      id="search-bar"
      onChange={
        [MockFunction] {
          "calls": Array [
            Array [
              null,
              "inception",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      placeholder="Search Movies"
      type="text"
    />,
    <button
      className="btn btn-primary"
      type="button"
    >
      Search
    </button>,
  ],
  "className": "search-bar",
}
`;
